### PR TITLE
Update to spec v0.11

### LINKED ISSUE #955: Update to Spec v0.11.x
## Description

Update `types`, `state_processing` and `beacon_node` for v0.11.x spec changes

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Closes #955

## Proposed Changes

* Update `types` and `state_processing` to v0.11.1

## TODO

- [x] Update to v0.11.1 when it is released (test vectors will be failing until then, but pass locally with rewards fix)
- [x] Update the `beacon_node` (in progress)
- [x] Update the validator client
- [x] Update HTTP API docs (for v0.11 and for new endpoint?)
- [x] Update deposit contract version tag (should be a no-op)
- [x] Update README spec tag

### CODE REVIEW

File: eth2/state_processing/src/per_block_processing/tests.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use types::test_utils::{
 use types::*;
 
 pub const NUM_DEPOSITS: u64 = 1;
-pub const VALIDATOR_COUNT: usize = 10;
+pub const VALIDATOR_COUNT: usize = 64;
```
Comment (michaelsproul) [inline eth2/state_processing/src/per_block_processing/tests.rs]: These tests were failing on the newly introduced empty attestation bitfield check, so I had to bump this

File: eth2/utils/eth2_testnet_config/src/lib.rs [hunk 2]
```diff
@@ -202,6 +202,7 @@ mod tests {
 
     type E = MainnetEthSpec;
 
+    /* TODO: disabled until testnet config is updated for v0.11
```
Comment (michaelsproul) [inline eth2/utils/eth2_testnet_config/src/lib.rs]: I'm not sure about the longevity of this test. It's tempting to delete it, or have some flag that turns "hardcoded testnet dirs" on/off?

Comment (michaelsproul) [reply eth2/utils/eth2_testnet_config/src/lib.rs]: Yeah sounds good, I definitely agree with keeping the feature, just thought maybe we should switch it off at certain points between releases, by the logic that code on `master` is only sometimes going to work with the hardcoded config. E.g. once this PR is merged, the code on master won't be useful for syncing with `testnet5`. Or we can just comment hop

Comment (michaelsproul) [reply eth2/utils/eth2_testnet_config/src/lib.rs]: (I'm happy to comment hop for now)

File: eth2/state_processing/src/per_block_processing.rs [hunk 3]
```diff
@@ -136,14 +136,26 @@ pub fn per_block_processing<T: EthSpec>(
...
     state: &mut BeaconState<T>,
     block: &BeaconBlock<T>,
     spec: &ChainSpec,
 ) -> Result<(), BlockOperationError<HeaderInvalid>> {
+    // Verify that the slots match
     verify!(block.slot == state.slot, HeaderInvalid::StateSlotMismatch);
 
+    // Verify that proposer index is the correct index
+    let proposer_index = block.proposer_index as usize;
+    let state_proposer_index = state.get_beacon_proposer_index(block.slot, spec)?;
```
[the remark this answers is not in the document]
Comment (michaelsproul) [reply eth2/state_processing/src/per_block_processing.rs]: Yeah, that was my thinking too. At the moment `get_beacon_proposer_index` is pretty quick

File: beacon_node/rest_api/src/beacon.rs [hunk 4]
```diff
@@ -497,6 +497,14 @@ pub fn get_genesis_time<T: BeaconChainTypes>(
     ResponseBuilder::new(&req)?.body(&beacon_chain.head()?.beacon_state.genesis_time)
 }
 
+/// Read the `genesis_validators_root` from the current beacon chain state.
+pub fn get_genesis_validators_root<T: BeaconChainTypes>(
+    req: Request<Body>,
+    beacon_chain: Arc<BeaconChain<T>>,
+) -> ApiResult {
+    ResponseBuilder::new(&req)?.body(&beacon_chain.head()?.beacon_state.genesis_validators_root)
```
[the remark this answers is not in the document]
Comment (michaelsproul) [reply beacon_node/rest_api/src/beacon.rs]: Good catch, I'll do the same for `genesis_time` while I'm there

File: eth2/types/src/chain_spec.rs [hunk 5]
```diff
@@ -112,53 +121,96 @@ pub struct ChainSpec {
...
+    ) -> Hash256 {
         let domain_constant = self.get_domain_constant(domain);
 
-        let mut bytes: Vec<u8> = int_to_bytes4(domain_constant);
-        bytes.append(&mut fork_version.to_vec());
-
-        let mut fork_and_domain = [0; 8];
-        fork_and_domain.copy_from_slice(&bytes);
+        let mut domain = [0; 32];
+        domain[0..4].copy_from_slice(&int_to_bytes4(domain_constant));
```
[the remark this answers is not in the document]
Comment (michaelsproul) [reply eth2/types/src/chain_spec.rs]: I changed `int_to_bytes4`

**Comment by michaelsproul** [comment]
All review comments addressed! :relaxed:

**Comment by hermanjunge** [comment]
:tada:

**Pull request merged**